server/services/player.ts

```diff
@@ -609,6 +612,29 @@ export default class PlayerService extends EventEmitter {
         throw new Error(`Unsupported specialist reward type: ${game.settings.technology.specialistTokenReward}.`);
     }
 
+    deductAllianceUpkeepCost(game, player, credits) {
+      const upkeepCosts = {
+        'none': 0,
+        'cheap': .02,
+        'standard': .05,
+        'expensive': .10
+      };
+
+      let costPerAlly = upkeepCosts[game.settings.alliances.allianceUpkeepCost];
+      if (!costPerAlly) {
+        return null;
```

Comment:
I fear we'll have bigger problems here if we get into a situation where the settings are out of sync.

We'll migrate existing data to ensure all games have a default setting, we can probably safely assume that this will never be null but if you insist, throw an actual error to break execution.

Reply:
I put that there to short circuit the logic if upkeep is zero.  Eg if costPerAlly == 0 then no need to do any further calculations.
Returning null will set the allianceUpkeepResult to null which will prevent the message from appearing in the event log (there is a check to see if it is defined in PlayerGalacticCycleCompleteEvent)  

I will change the check to == 0 to be more explicit.